dask/array/rechunk.py

```diff
@@ -430,6 +476,8 @@ def plan_rechunk(old_chunks, new_chunks, itemsize,
         The maximum block size (in bytes) we want to produce during an
         intermediate step
     """
+    # TODO: revisit this for unknown chunks. Is this still correct?
+    # Do we have to worry about the memory sizes not being respected?
```

Comment:
You should, unless you don't care about an intermediate result possibly swamping memory and swapping out to disk. This is why I suggested that the current heuristic may not be correct for unknown-sized arrays.